aptos-move/aptos-vm/src/aptos_vm.rs

```diff
@@ -214,30 +215,66 @@ impl AptosVM {
         }
     }
 
-    fn success_transaction_cleanup<S: MoveResolverExt>(
+    fn success_transaction_cleanup<S: MoveResolverExt + StateView>(
         &self,
-        mut session: SessionExt<S>,
+        storage: &S,
+        user_txn_change_set_ext: ChangeSetExt,
         gas_meter: &mut AptosGasMeter,
         txn_data: &TransactionMetadata,
         log_context: &AdapterLogSchema,
     ) -> Result<(VMStatus, TransactionOutputExt), VMStatus> {
+        let storage_with_changes =
+            DeltaStateView::new(storage, user_txn_change_set_ext.write_set());
+        let delta_write_set_mut = user_txn_change_set_ext
```

Comment:
Note: this is where I'm materializing the delta change set.

cc @georgemitenkov @gelash

Reply:
Why converting to VMStatus::Error(StatusCode::UNKNOWN_INVARIANT_VIOLATION_ERROR))? try_into_write_set_mut already returns Result<,.., VMStatus>?